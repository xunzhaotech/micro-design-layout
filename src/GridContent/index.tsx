import './index.less';

import { FunctionalComponent, CSSProperties } from 'vue';
import { PureSettings } from '../config/defaultSettings';
import { useRouteContext } from '../RouteContext';

export interface GridContentProps {
  contentWidth?: PureSettings['contentWidth'];
  prefixCls?: string;
  style?: CSSProperties;
}

const GridContent: FunctionalComponent<GridContentProps> = (props, { slots }) => {
  const { contentWidth, getPrefixCls } = useRouteContext();
  const customPrefixCls = props.prefixCls || getPrefixCls();
  const customContentWidth = props.contentWidth || contentWidth;
  return (
    <div
      class={{
        [`${customPrefixCls}-grid-content`]: true,
        wide: customContentWidth === 'Fixed',
      }}
    >
      <div class={`${customPrefixCls}-grid-content-children`}>{slots.default?.()}</div>
    </div>
  );
};

export default GridContent;
