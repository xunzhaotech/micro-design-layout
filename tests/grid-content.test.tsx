import { mount } from '@vue/test-utils';
import GridContent from '../src/GridContent';

describe('GridContent', () => {
  it('shoul render with empty children', () => {
    const wrapper = mount(GridContent, {});
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shoul render with prop Fixed', () => {
    const wrapper = mount({
      setup() {
        return () => <GridContent contentWidth="Fixed"></GridContent>;
      },
    });
    expect(wrapper.classes().indexOf('wide') > -1).toBe(true);
  });

  it('shoul render with children', () => {
    const wrapper = mount({
      setup() {
        return () => (
          <GridContent contentWidth="Fixed">
            <div>children</div>
          </GridContent>
        );
      },
    });
    expect(
      wrapper.find('.ant-pro-grid-content-children').element.innerHTML === '<div>children</div>',
    ).toBe(true);
  });
});
