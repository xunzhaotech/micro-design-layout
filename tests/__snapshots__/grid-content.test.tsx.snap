// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContent shoul render with empty children 1`] = `
<div class="ant-pro-grid-content">
  <div class="ant-pro-grid-content-children">
    <!---->
  </div>
</div>
`;
